docs/cli.rst

```diff
@@ -50,9 +50,10 @@ Config values can be read via the built-in :ref:`config_dependency_provider` dep
 
 Environment variable substitution
 ---------------------------------
-YAML configuration files have basic support for environment variables.
+YAML configuration files have support for environment variables.
 You can use bash style syntax: ``${ENV_VAR}``
 Optionally you can provide default values ``${ENV_VAR:default_value}``
+default values can contains environment variables recursively ``${ENV_VAR:default_${OTHER_ENV_VAR:value}}`` (note: this feature require regex package)
```

Comment:
This seems to be a new sentence. Maybe we should start it with capital letter and end with period? Looks like 2 sentences above yours missing period as well.